File: py/Elementary/digits_multiplication.py
#!/usr/bin/env checkio --domain=py run digits-multiplication

# https://py.checkio.org/mission/digits-multiplication/

# You are given a positive integer.    Your function should calculate the product of the digits excluding any zeroes.
# 
# For example: The number given is 123405. The result will be 1*2*3*4*5=120 (don't forget to exclude zeroes).
# 
# Input:A positive integer.
# 
# Output:The product of the digits as an integer.
# 
# Precondition:0 < number < 106
# 
# 
# END_DESC

def checkio(number: int) -> int:
    return eval("*".join([x for x in str(number).replace("0","")]))

File: js/Elementary/digits_multiplication.js
#!/usr/bin/env checkio --domain=js check digits-multiplication
// https://js.checkio.org/mission/digits-multiplication/

"use strict";

function digitsMultip(data) {
    // "1" concat number to transfer number to string
    // -> remove all 0
    // -> split to array with single digital
    // -> calculate product, the "*" will auto transfer string to number
    return ("1" + data).replace(/0/g, "").split("").reduce((a, b) => a * b);
}

var assert = require('assert');

if (!global.is_checking) {
    assert.equal(digitsMultip(123405), 120, "1st");
    assert.equal(digitsMultip(999), 729, "2nd");
    assert.equal(digitsMultip(1000), 1, "3rd");
    assert.equal(digitsMultip(1111), 1, "4th");
    console.log("Coding complete? Click 'Check' to review your tests and earn cool rewards!");
}